pkg/apis/experimental/types.go

```diff
@@ -508,35 +509,56 @@ type IngressStatus struct {
 	LoadBalancer api.LoadBalancerStatus `json:"loadBalancer,omitempty"`
 }
 
-// IngressRule represents the rules mapping the paths under a specified host to the related backend services.
-type IngressRule struct {
+// IngressHost represents the rules mapping the paths under a specified host to
+// the related backend services.
+type IngressHost struct {
 	// Host is the fully qualified domain name of a network host, or its IP
 	// address as a set of four decimal digit groups separated by ".".
-	// Conforms to RFC 1738.
+	// Conforms to RFC 1738. Incoming requests are matched against the Host
+	// before the IngressRule.
 	Host string `json:"host,omitempty"`
+	// IngressRule represents a rule to route requests for this IngressHost.
+	IngressRule `json:",inline"`
+}
+
+// IngressRule represents a rule to apply against incoming requests. If the
+// rule is satisfied, the request is routed to the specified backend.
+type IngressRule struct {
+	// Currently mixing different types of rules in a single Ingress is
+	// disallowed, so exactly one of the following must be set.
+	HTTP *HTTPIngressRule `json:"http"`
```

Comment:
Hmm, I see what you're saying. Where would ingress specific global information go, into each path (or the claim, but putting loadbalancer configuration into a claim feels messy)? an example of such state is `option forwardfor` which enables insertion of X-Forwarded-For, or `option http-keep-alive` which does what you'd think (allows client server keepalive). 

I actually wanted to go the opposite direction and give the HTTPIngressRule Annotations.